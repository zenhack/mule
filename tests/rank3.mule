#XFAIL?
let rank3 : all b. ((all a. a -> a) -> b) -> b =
  fn rank2. rank2 (fn x. x)
in
rank3 (fn rank1. rank1 (rank1 4))
