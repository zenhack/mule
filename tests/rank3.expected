
Desugared:

	(letrec (types ())
	 (values
	  ((rank3
	    ((: (all b (((all a (a -> a)) -> b) -> b))) (fn rank2 (rank2 (fn x x)))))))
	 (rank3 (fn rank1 (rank1 (rank1 (Integer 4))))))

inferred type:

	int

Runtime term:

	(LetRec
	 (((0 (Lam (0 () (App ((Var 0) (Lam (0 () (Var 0)))))))))
	  (App
	   ((Var 0) (Lam (0 () (App ((Var 0) (App ((Var 0) (Const (Integer 4))))))))))))

Evaluated:

	(Const (Integer 4))
